### PR TITLE
As requested, here is my code

### PR DESCRIPTION
**Pull request by Bongo50**
For [tardis.fandom.com](https://tardis.fandom.com/) rather then leaguepedia.

### CODE REVIEW

**Comment by Bongo50** [comment]
I forgot to mention it in the description but, while these modules may end up being updated before rollout on-wiki, the bits that deal with the storage and retrieval of the data won't.

File: storyCite.lua [hunk 1]
```diff
@@ -0,0 +1,248 @@
+local util_vars = require('Module:VarsUtil') --from https://thealchemistcode.fandom.com/wiki/Module:VarsUtil
+local util_link = require('Module:LinkUtil')
+local cache = require('mw.ext.LuaCache')
+local PREFIX = 'SC_CACHE_01-'
+local dump
+local p = {} --p stands for package
+local h = {} --h stands for helper
+
+-- from https://stackoverflow.com/a/7615129
+function h.splitString (inputstr, sep)
```
Comment (RheingoldRiver) [inline storyCite.lua:10]: put this into a separate util module called StringUtil or TextUtil, and then just call it `split` (you can copy mine if you want, it's at TextUtil)

File: storyCite.lua [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		if sep == nil then
+				sep = "%s"
```
Comment (RheingoldRiver) [inline storyCite.lua:12]: I personally default to comma. If you're gonna default to whitespace, I'd default to `%s*`

File: storyCite.lua [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+		bin = cache.set(PREFIX .. name, outputText)
+		bin = mw.smw.set("Story info", outputText) --documentation for this is down so I guessed the syntax. It doesn't seem to work.
+	end
+end
+
+function h.getInfo(story)
+	local info 
+	local bin --used to store the output of storing functions
+	if util_vars.getVar(PREFIX .. story) then
+		info = util_vars.getVar(PREFIX .. story)
```
Comment (RheingoldRiver) [inline storyCite.lua:200]: just return now, no need to keep doing anything

File: storyCite.lua [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+		bin = util_vars.setVar(PREFIX .. name, outputText)
+		bin = cache.delete(PREFIX .. name)
+		bin = cache.set(PREFIX .. name, outputText)
+		bin = mw.smw.set("Story info", outputText) --documentation for this is down so I guessed the syntax. It doesn't seem to work.
+	end
+end
+
+function h.getInfo(story)
+	local info 
+	local bin --used to store the output of storing functions
```
Comment (RheingoldRiver) [inline storyCite.lua:198]: this isn't necessary; you don't have to set this in the outer scope unless you're going to use it in the outer scope. just say `local bin =` within the scope you need it later.

also, if you aren't going to use the output that a function returns, it's not required to store it into a variable. so you can just say `util_vars.setVar(PREFIX .. story, info)` without anything else on the line.

File: storyCite.lua [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+	if util_vars.getVar(PREFIX .. story) then
+		info = util_vars.getVar(PREFIX .. story)
+	elseif cache.get(PREFIX .. story) then
+		info = cache.get(PREFIX .. story)
+		bin = util_vars.setVar(PREFIX .. story, info)
+	elseif mw.smw.ask( story .. "\n?Story info") then
+		info = mw.smw.ask( story .. "\n?Story info")
+		bin = util_vars.setVar(PREFIX .. story, info)
+		bin = cache.delete(PREFIX .. name)
```
Comment (RheingoldRiver) [inline storyCite.lua:207]: you don't have to delete anything; we only got here if there was for certain no cache entry in the first place

File: storyCite.lua [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+function h.getInfo(story)
+	local info 
+	local bin --used to store the output of storing functions
+	if util_vars.getVar(PREFIX .. story) then
+		info = util_vars.getVar(PREFIX .. story)
+	elseif cache.get(PREFIX .. story) then
+		info = cache.get(PREFIX .. story)
+		bin = util_vars.setVar(PREFIX .. story, info)
+	elseif mw.smw.ask( story .. "\n?Story info") then
```
Comment (RheingoldRiver) [inline storyCite.lua:204]: did you check if this module works without any of the cache part entered? (e.g. I think you need `("[[%s]]"):format(story)` here? not sure how mw.smw.ask works though)

File: storyCite.lua [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+function p.displayCitation(frame)
+	local story = frame:getParent().args[1]
+	return "''[[" .. story .. "|" .. util_link.stripDab(story) .. "]]'' <sup><span class=\"mw-customtoggle-{{#vardefineecho:id|{{#expr:{{#var:id|0}}+1}}}}\">+</span> <span class=\"mw-collapsible mw-collapsed\" id=\"mw-customcollapsible-{{#var:id}}\">" .. h.getInfo(story) .. "</span></sup>"
+end
+
+--[[for testing that data is being collected.
+	from https://sandbox.semantic-mediawiki.org/wiki/Module:Smw
+	This dumps the variable (converts it into a string representation of itself)--]]
+dump = function(entity, indent, omitType)
```
Comment (RheingoldRiver) [inline storyCite.lua:223]: this should be in a separate module (in fact when the smw wiki is back up you should just copy that entire module here as Module:Smw)

File: storyCite.lua [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+		bin = cache.set(PREFIX .. story, info)
+	else
+		info = "No data."
+	end
+	return info
+end
+
+function p.displayCitation(frame)
+	local story = frame:getParent().args[1]
+	return "''[[" .. story .. "|" .. util_link.stripDab(story) .. "]]'' <sup><span class=\"mw-customtoggle-{{#vardefineecho:id|{{#expr:{{#var:id|0}}+1}}}}\">+</span> <span class=\"mw-collapsible mw-collapsed\" id=\"mw-customcollapsible-{{#var:id}}\">" .. h.getInfo(story) .. "</span></sup>"
```
Comment (RheingoldRiver) [inline storyCite.lua:217]: this needs a frame:preprocess i think?

File: storyCite.lua (hunk 7 above)
Comment (RheingoldRiver) [reply storyCite.lua:223]: jk it's here: https://yugioh.fandom.com/wiki/Module:SMW (I just googled `Module:SMW`)

File: storyCite.lua [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+		end
+		
+		-- combines {{{adapted from}}} and {{{novelisation of}}}
+		if args["adapted from"] and args["novelisation of"] then
+			args["adapted from"] = args["adapted from"] .. ", " .. args["novelisation of"]
+		elseif (not args["adapted from"]) and args["novelisation of"] then
+			args["adapted from"] = args["novelisation of"]
+		end
+		
+		--gets and formats data for items that are unlinked	in the infobox
```
Comment (RheingoldRiver) [inline storyCite.lua:45]: put this into a function called h.getAndFormatLinkedItems

File: storyCite.lua [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+					elseif i == #current then
+						data[key] = data[key]
+					else
+						data[key] = data[key] .. ", "
+					end
+				end
+			end
+		end
+		
+		--same concept as above but doesn't add links
```
Comment (RheingoldRiver) [inline storyCite.lua:63]: h.getAndFormatUnlinkedItems

then util_table.merge() the results of that into your data from the previous function

(see my TableUtil for `util_table.merge()`)

File: storyCite.lua [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+					data["issues"] = data["issues"] .. " and "
+				elseif i == #current then
+					data["issues"] = data["issues"]
+				else
+					data["issues"] = data["issues"] .. ", "
+				end
+			end
+		end
+		
+		-- making series text
```
Comment (RheingoldRiver) [inline storyCite.lua:79]: `data.series = h.getAndFormatSeriesText(args)`

File: storyCite.lua [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+				data["series"] = args["series"]
+			end
+			if args["season number"] then
+				data["series"] = data["series"] .. " " .. "[[" .. util_link.stripPipe(args["season number"]) .. "|" .. util_link.stripPipe(util_link.stripDab(args["season number"])) .. "]]"
+			end
+		else
+			data["series"] = ""
+		end
+		
+		-- making release year text
```
Comment (RheingoldRiver) [inline storyCite.lua:97]: `data["release year"] = h.getReleaseYearText(args)`

File: storyCite.lua [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+				end
+				data["release year"] = data["release year"] .. " the years [[" .. releaseYear .. " (releases)|" .. releaseYear .."]] and [[" .. releaseEndYear .. " (releases)|" .. releaseEndYear .. "]]"
+			else
+				data["release year"] = "in the year [[" .. releaseYear .. " (releases)|" .. releaseYear .."]]"
+			end
+		else
+			data["release year"] = ""
+		end
+		
+		-- making outputText
```
Comment (RheingoldRiver) [inline storyCite.lua:129]: move all of this below the smw store stuff and then `return h.makeOutput(data, args)`

File: storyCite.lua (hunk 8 above)
Comment (Bongo50) [reply storyCite.lua:217]: It works fine so I don't think it does.

**Pull request merged**